src/dht.js

```diff
@@ -1,43 +1,82 @@
 'use strict'
 
-const nextTick = require('async/nextTick')
 const errCode = require('err-code')
-const promisify = require('promisify-es6')
 
 const { messages, codes } = require('./errors')
 
-module.exports = (node) => {
-  return {
-    put: promisify((key, value, callback) => {
-      if (!node._dht) {
-        return nextTick(callback, errCode(new Error(messages.DHT_DISABLED), codes.DHT_DISABLED))
-      }
+module.exports = (node, DHT, config) => {
+  const dht = new DHT({
+    dialer: {
+      dial: (peer, options) => node.dial(peer, options),
+      dialProtocol: (peer, protocols, options) => {
+        const recordedPeer = node.peerStore.get(peer.toB58String())
```

Comment:
When the dial is happening, if we try to `dial` using the `peer-id`, the `peer-info` stored gets replaced, removing the `multiaddrs` that previously existed:

https://github.com/libp2p/js-libp2p/blob/refactor/async-await/src/get-peer-info.js#L36-L41 

I think we can tackle this once we get rid of `peer-info`